File: app/src/main/java/com/antyzero/smoksmog/ui/BaseActivity.java
package com.antyzero.smoksmog.ui;

import android.app.Activity;
import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;

import com.antyzero.smoksmog.R;
import com.trello.rxlifecycle.components.support.RxAppCompatActivity;

import butterknife.ButterKnife;
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;

public abstract class BaseActivity extends RxAppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initOnCreate(this);
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        ButterKnife.bind(this);
    }

    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        ButterKnife.bind(this);
    }

    @Override
    public void setContentView(View view, ViewGroup.LayoutParams params) {
        super.setContentView(view, params);
        ButterKnife.bind(this);
    }

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
    }

    /**
     * Shared initialization among activities, use it you cannot extend BaseActivity
     *
     * @param activity for access to various data
     */
    public static void initOnCreate(Activity activity) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            int color;
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                color = activity.getResources().getColor(R.color.primary, activity.getTheme());
            } else {
                //noinspection deprecation
                color = activity.getResources().getColor(R.color.primary);
            }
            activity.getWindow().setNavigationBarColor(color);
        }
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/BaseActivity.kt
package com.antyzero.smoksmog.ui

import android.app.Activity
import android.content.Context
import android.os.Build
import com.antyzero.smoksmog.R
import com.antyzero.smoksmog.dsl.getCompatColor
import com.trello.rxlifecycle.components.support.RxAppCompatActivity
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper

abstract class BaseActivity : RxAppCompatActivity() {

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase))
    }

    companion object {

        /**
         * Shared initialization among activities, use it you cannot extend BaseActivity

         * @param activity for access to various data
         */
        fun initOnCreate(activity: Activity) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                activity.window.navigationBarColor = activity.getCompatColor(R.color.primary)
            }
        }
    }
}
